website/project/model.py

```diff
@@ -2609,3 +2738,137 @@ def to_json(self):
                       for x in self.nodes if not x.is_deleted],
             "anonymous": self.anonymous
         }
+
+
+def validate_retraction_state(value):
+    acceptable_states = ['pending', 'retracted', 'cancelled']
```

Comment:
The valid states should be named constants. They can be defined on the `Retraction` class. That way, we can check `Retraction.PENDING` instead of relying on "magic" strings.